assignment7.txt

```diff
@@ -0,0 +1,22 @@
+
+
+1. ++*p increments what the pointer points to whereas the *p++ actually increases the value of the pointer. *++p is also increases what the pointer points to.
+
+
+2. Oh the left is guaanteed because its higher in the order of operations.
+
+
+3. pointers allow easier use with arrays. You can go through the pointer without the for loop and it allows easier access outside of the main function. 
+
+
+4.1 char array 
+4.2 Invalid becaus xyz in quotes is not an arry
+4.3 Invalid becuse that will never be true
+4.4 pointer
+4.5 address is 0
```

Comment:
This actually returns the type int*. Recall that a[0] would be of type int and the ampersand just returns the memory address. So &a[0] would return the pointer to the address (since a pointer is basically an address, which is why all pointers are 8 bytes no matter what they're pointing to), which would be of type int*.